test/extended/clusterup.sh

```diff
@@ -0,0 +1,242 @@
+#!/bin/bash
+source "$(dirname "${BASH_SOURCE}")/../../hack/lib/init.sh"
+
+# Allow setting $JUNIT_REPORT to toggle output behavior
+if [[ -n "${JUNIT_REPORT:-}" ]]; then
+    export JUNIT_REPORT_OUTPUT="${LOG_DIR}/raw_test_output.log"
+fi
+
+os::util::ensure::built_binary_exists 'oc'
+os::util::environment::setup_all_server_vars "test-extended/clusterup"
+
+function os::test::extended::clusterup::run_test () {
+    local test="${1}"
+    local funcname="os::test::extended::clusterup::${test}"
+    local global_home="${HOME}"
+    os::log::info "starting test -- ${test}"
+
+    os::test::junit::declare_suite_start "extended/clusterup/${test}"
+
+    local test_home="${ARTIFACT_DIR}/${test}/home"
+    mkdir -p "${test_home}"
+    export HOME="${test_home}"
+    pushd "${HOME}" &> /dev/null
+    os::log::info "Using ${HOME} as home directory"
+    ${funcname}
+    popd &> /dev/null
+    export HOME="${global_home}"
+
+    os::log::info "cluster up -- ${test}: ok"
+    os::test::junit::declare_suite_end
+}
+
+function os::test::extended::clusterup::verify_router_and_registry () {
+    os::cmd::expect_success "oc login -u system:admin"
+    os::cmd::expect_success_and_text "oc get svc -n default" "docker-registry"
+    os::cmd::expect_success_and_text "oc get svc -n default" "router"
+    os::cmd::try_until_text "oc get endpoints docker-registry -o jsonpath='{ .subsets[*].ports[*].port }' -n default" "5000" $(( 10*minute )) 1
```

Comment:
This has the chance of matching port `50000` and `15000` right? That's why the `jsonpath` I suggested grabbed a specific port entry